### PR TITLE
Temporary documents

### PR DESCRIPTION
**Pull request by ganler**
Temporary markdown documents about installation, quickstart, etc.

For more documentation, we are trying to deploy it on read-the-docs.

@luomai @lgarithm

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,33 +9,30 @@
...
+
+Markdown documents are available in `docs/markdown`.
+
+## Getting Started
+
+### Prediction
+
+> For more detailed guidance of installation/examples(more examples), please refer to [QuickStart | Prediction](docs/markdown/quick_start/prediction.md) and [Installation | Prediction](docs/markdown/install/prediction.md).
+
+Install CUDA Toolkit, TensorRT, OpenCV and gFlags(cmake version), and make sure a C++17 compiler is available on your system.
```
Comment (luomai) [inline README.md:24]: This is too vague. We need a clear version requirement for these libraries.

File: README.md [hunk 2]
```diff
@@ -9,33 +9,30 @@
...
+
+```bash
```
Comment (luomai) [inline README.md:26]: @lgarithm Could you have a look at this?

File: README.md [hunk 3]
```diff
@@ -9,33 +9,30 @@
...
-    - ONNX.
-    - Cuda Engine Protobuf.
-  - Good performance
+## Documentation
+
+[Documentation](Coming Soon)
+
+Markdown documents are available in `docs/markdown`.
+
+## Getting Started
```
Comment (luomai) [inline README.md:18]: Where is the training part？

Comment (ganler) [reply README.md:18]: @Gyx-One is working on it.

**Comment by ganler** [comment]
@lgarithm Hi, I was trying to enable exporting a shared library(for tutorial) but failed to include the ttl headers. Do you have some suggestions for that? Or we just let users use it as a static library and install via ExternalProject(cmake)?

**Comment by ganler** [comment]
> @lgarithm Hi, I was trying to enable exporting a shared library(for tutorial) but failed to include the ttl headers. Do you have some suggestions for that? Or we just let users use it as a static library and install via ExternalProject(cmake)?

Now I enabled to install via ExternalProject. But we need to merge this PR. (Cause we didn't set THIRDPARTY_PREFIX a option)

**Comment by ganler** [comment]
I met linking problem when doing ExternalProject... This is strange...

```cmake
CMAKE_MINIMUM_REQUIRED(VERSION 3.5)

PROJECT(PlayWithHyperPose)

SET(CMAKE_CXX_STANDARD 17)
ADD_EXECUTABLE(${PROJECT_NAME} main.cpp)

# Cuda, CuDNN, TensorRT
IF(IS_DIRECTORY ${CUDA_TOOLKIT_ROOT_DIR})
    MESSAGE("Using CUDA_TOOLKIT_ROOT_DIR=${CUDA_TOOLKIT_ROOT_DIR}")
    INCLUDE_DIRECTORIES(SYSTEM ${CUDA_TOOLKIT_ROOT_DIR}/include)
    LINK_DIRECTORIES(SYSTEM ${CUDA_TOOLKIT_ROOT_DIR}/lib64)
ELSE()
    FIND_PACKAGE(CUDA REQUIRED)
    INCLUDE_DIRECTORIES(SYSTEM ${CUDA_INCLUDE_DIRS})
    EXECUTE_PROCESS(
        COMMAND arch
        COMMAND tr -d '\n'
        OUTPUT_VARIABLE ARCH)
    SET(CUDA_RT /usr/local/cuda/targets/${ARCH}-linux)

    LINK_DIRECTORIES(${CUDA_RT}/lib)
ENDIF()

# OpenCV
FIND_PACKAGE(OpenCV REQUIRED)

# HyperPose
INCLUDE(ExternalProject)
SET(HYPERPOSE_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/3rdparty)
SET(HYPERPOSE_GIT_URL https://github.com/tensorlayer/hyperpose.git)

EXTERNALPROJECT_ADD(
    hyperpose-repo
    LOG_DOWNLOAD ON
    LOG_INSTALL ON
    LOG_CONFIGURE ON
    GIT_REPOSITORY ${HYPERPOSE_GIT_URL}
    GIT_TAG 53b57c0ef982261b9f3917d8eb88f37b205fce90
    PREFIX ${HYPERPOSE_PREFIX}
    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${HYPERPOSE_PREFIX} -DTHIRDPARTY_PREFIX=${HYPERPOSE_PREFIX} -DBUILD_TESTS=OFF -DBUILD_EXAMPLES=OFF)

ADD_DEPENDENCIES(${PROJECT_NAME} hyperpose-repo)

INCLUDE_DIRECTORIES(${HYPERPOSE_PREFIX}/include)
find_library(HYPERPOSE_LIBRARIES hyperpose HINTS ${HYPERPOSE_PREFIX}/lib)

# Linking
TARGET_LINK_LIBRARIES(${PROJECT_NAME}  
	cudnn
        cudart
        nvinfer
        nvparsers
        nvonnxparser
	${HYPERPOSE_LIBRARIES}
        ${OpenCV_LIBS})
```

**Comment by ganler** [comment]
It seems that the doxygen link is broken...
https://openpose-plus.readthedocs.io/en/refactorization/markdown/apis/prediction.html

**Comment by lgarithm** [comment]
Let’s make basic features (e.g. more unit tests, better public header dependency) working before providing more fancy features like exporting shared library.


On 1 Jun 2020, at 4:50 PM, Jiawei Liu <notifications@github.com<mailto:notifications@github.com>> wrote:



@lgarithm<https://github.com/lgarithm> Hi, I was trying to enable exporting a shared library(for tutorial) but failed to include the ttl headers. Do you have some suggestions for that? Or we just let users use it as a static library and install via ExternalProject(cmake)?

**Pull request merged**